### PR TITLE
s3 compatible object storage support

### PR DESCRIPTION
**Pull request by k-anshul**

### CODE REVIEW

**Comment by k-anshul** [comment]
closes https://github.com/rilldata/rill-developer/issues/1691

File: runtime/connectors/s3/s3.go [hunk 1]
```diff
@@ -117,6 +118,15 @@ func (c connector) ConsumeAsFiles(ctx context.Context, env *connectors.Env, sour
...
 
 func getAwsSessionConfig(ctx context.Context, conf *Config, bucket string) (*session.Session, error) {
+	// if S3CompatibleEndpoint is set then fs support s3 API
+	if len(conf.S3CompatibleEndpoint) > 0 {
+		return session.NewSession(&aws.Config{
+			Region:           aws.String("us-east-1"),
+			Endpoint:         &conf.S3CompatibleEndpoint,
+			S3ForcePathStyle: aws.Bool(true),
+		})
+	}
```
Comment (begelundmuller) [inline runtime/connectors/s3/s3.go:128]: Why does region need to be set when an endpoint is passed?

Also, what about credentials for S3 compatible endpoints? Should people use the AWS environment variables to override?

**Review by begelundmuller — CHANGES_REQUESTED** [review]
Just fixed another bug in this code, sorry for the merge conflict

File: runtime/connectors/s3/s3.go (hunk 1 above)
Comment (k-anshul) [reply runtime/connectors/s3/s3.go:128]: The AWS SDK complains if region is not set. Cloudflare also allows user to set us-east-1 region for bwd compatibility reasons. 
Credentials work same like actual s3. I left some details here : https://github.com/rilldata/rill-developer/issues/1691#issuecomment-1415423446

File: runtime/connectors/s3/s3.go [hunk 2]
```diff
@@ -59,6 +59,7 @@ type Config struct {
 	GlobMaxObjectsMatched int    `mapstructure:"glob.max_objects_matched"`
 	GlobMaxObjectsListed  int64  `mapstructure:"glob.max_objects_listed"`
 	GlobPageSize          int    `mapstructure:"glob.page_size"`
+	S3CompatibleEndpoint  string `mapstructure:"s3_compatible_endpoint"`
```
[the remark this answers is not in the document]
Comment (k-anshul) [reply runtime/connectors/s3/s3.go:62]: Lets make this just endpoint in yaml but S3Endpoint in actual code since the configs are common.

**Review by begelundmuller — APPROVED** [review]

**Review by begelundmuller — APPROVED** [review]

**Comment by begelundmuller** [comment]
Made a little change to use the configured `region` if set, although I'm not sure it will ever make a difference. Everything else looks good, merging now

**Pull request merged**